File: src/sbt-test/plugin-tests/scala9-sources-test/src/main/scala-jdk9/com/example/MyJavaVersionDependentImpl.scala
package com.example

import scala.Console

import java.io.File
import java.io.IOException

class MyJavaVersionDependentImpl {
  def print(): Unit = {
    println(Console.GREEN + "Class from `java9`, while running Java " + System.getProperty("java.version") + Console.RESET)
  }
}

File: src/java11-sources-test/src/main/java/com/example/MyJavaVersionDependentImpl.java
package com.example;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class MyJavaVersionDependentImpl {
  public void print() throws IOException {
    System.out.println("Class from `java{6,7,8}`, while running [Java " + System.getProperty("java.version") + "]");
  }
}
